lib/driver.c

```diff
@@ -241,6 +241,19 @@ log_src_driver_free(LogPipe *s)
 
 /* LogDestDriver */
 
+static LogQueue *
+_create_log_queue(LogDestDriver *self, const gchar *persist_name)
+{
+  GlobalConfig *cfg = log_pipe_get_config(&self->super.super);
+
+  gint log_fifo_size = self->log_fifo_size < 0 ? cfg->log_fifo_size : self->log_fifo_size;
+
+  if (cfg_is_config_version_older(cfg, 0x0316))
+    return log_queue_fifo_legacy_new(log_fifo_size, persist_name);
```

Comment:
[optional] Could we in this case also provide some warning ?

Reply:
Good idea. We should probably do that somewhere else or use `warning_once`, because we don't want to repeat the same warning for every worker and destination.